### DISCUSSION TITLE
Get current time step

### DISCUSSION THREAD

**Discussion by Wang12123w** [idaholab/moose#25304] [Q&A General]
Hi,
I have a simple question. I am writing code that performs an action when the current time step exceeds a certain value. How can I obtain the current time step?
Cheers!
Wang

**Comment by GiudGiud** [comment]
Hello

From the input file, that would be done using a postprocessor. i dont see one that does it though.

From the code, you can get the current time step from the FEProblem object that moose objects have as a class attribute
https://mooseframework.inl.gov/docs/doxygen/moose/classFEProblemBase.html#a01bc321edf2887a0daf998d7abc49dd3

Guillaume
